x-pack/plugins/ml/public/explorer/explorer_swimlane.js

```diff
@@ -0,0 +1,472 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+
+/*
+ * React component for rendering Explorer dashboard swimlanes.
+ */
+
+//import PropTypes from 'prop-types';
+import React from 'react';
+
+import _ from 'lodash';
+import d3 from 'd3';
+import $ from 'jquery';
+import moment from 'moment';
+
+// don't use something like plugins/ml/../common
+// because it won't work with the jest tests
+import { numTicksForDateFormat } from '../util/chart_utils';
+import { getSeverityColor } from '../../common/util/anomaly_utils';
+import { mlEscape } from '../util/string_utils';
+import { mlChartTooltipService } from '../components/chart_tooltip/chart_tooltip_service';
+
+export class ExplorerSwimlane extends React.Component {
+  static propTypes = {
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      cellMouseoverActive: true
+    };
+  }
+
+  componentWillUnmount() {
+    const { mlExplorerDashboardService } = this.props;
+    mlExplorerDashboardService.dragSelect.unwatch(this.dragSelectListener);
+
+  }
+  componentDidMount() {
+    const element = $(this.rootNode).parent();
+
+    const {
+      appState,
+      mlExplorerDashboardService,
+      swimlaneData,
+      swimlaneType
+    } = this.props;
+
+    // Consider the setting to support to select a range of cells
+    if (!mlExplorerDashboardService.allowCellRangeSelection) {
+      element.addClass('ml-hide-range-selection');
+    }
+
+    // Listen for dragSelect events
+    const that = this;
+    this.dragSelectListener = function ({ action, elements = [] }) {
```

Comment:
if these were converted to arrow functions you should mean the assignment of `that` could be removed.
`cellClick` could also be made a method and an arrow function.
probably not for this PR, but maybe a future refactor